wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,172 @@
+  using control-plane APIs to help coordinate maintenance and drain a Node.
+- To support the new features, node maintenance, scheduler, descheduler, pod autoscaling, kubelet,
+  and other actors should use a new eviction API to gracefully remove pods. This would enable new
+  migration strategies that prefer to surge (upscale) pods first rather than downscale them. It
+  would also allow other users/components to monitor pods that are gracefully removed/terminated
+  and provide better behaviour in terms of de/scheduling, scaling and availability.
+- As an end user, I would like more alternatives to blue-green upgrades, especially with special
+  hardware accelerators; it's far too expensive. I would like to choose a strategy on how to
+  coordinate the node drain and the upgrade to achieve better cost-effectiveness.
+- As a cloud provider, I need to perform regular maintenance on the hardware in my fleet. Enhancing
+  Kubernetes to help CSPs safely remove hardware will reduce operational costs.
+- As an end user or admin, I would like to use a mixture of on-demand and temporary spot instances
+  in my clusters to reduce cloud expenditure. Having more reliable lifecycle and drain mechanisms
+  for nodes will improve cluster stability in scenarios where instances may be terminated by the
+  cloud provider due to cost-related thresholds.
+- As a user, I want to prevent any disruption to my pet or expensive workloads (VMs, ML with
+  accelerators) and either prevent termination for a long period of time or have a reliable
+  migration path. Features like `terminationGracePeriodSeconds` are not sufficient as the
+  termination/migration can take hours if not days.
+- As a user, I want my application to finish all network and storage operations before terminating a
+  pod. This includes closing pod connections, removing pods from endpoints, writing cached writes
+  to the underlying storage and completing storage cleanup routines.
+
+## Deliverables
+
+The WG will coordinate requirement gathering and design, eventually leading to implementation.
+
+Area we expect to explore:
+
+- An API to express node drain/maintenance.
+- An API to solve the problems wrt the API-initiated Eviction API and PDBs.
+- An API/mechanism to gracefully terminate pods during a node shutdown.
+- An API to deschedule pods that use DRA devices.
+- An API to remove pods from endpoints before they terminate.
+- An API to track the schedulability (e.g. readiness) and capabilities of the node.
+- Introduce enhancements across multiple Kubernetes SIGs to add support and integration for the new
+  APIs to solve wide range of issues.
+
+We expect to provide reference implementations of the new APIs including but not limited to
+controllers, API validation, integration with existing core components and extension points for the
+ecosystem. This should be accompanied by E2E / Conformance tests.
+
+## Relevant KEPs and Documents
```

Comment:
How about "Node Resource Hot Plug"? https://github.com/kubernetes/enhancements/issues/3953

Reply:
Ack, good to discuss. Although I am not sure how much intersection there is.

Reply:
Thanks for suggesting and adding it here.